src/meteors/shap/explanation.py

```diff
@@ -0,0 +1,189 @@
+
+    Args:
+        data (np.ndarray): a numpy array containing the input data.
+        explanations (np.ndarray): a numpy array containing the SHAP explanations.
+        If the model outputs a single value, the shape of the array should be equal to the shape of the input data.
+        In case the model outputs multiple values, the last dimension of the array should be equal to the number of outputs and the rest of the dimensions should be equal to the input data.
+        explanation_method (str): the method used to generate the explanation.
+        feature_names (list[str]): a list of feature names.
+        aggregations (list[list[str]] | list[str]): a list of feature aggregations.
+    """
+
+    data: Annotated[
+        np.ndarray,
+        BeforeValidator(ensure_data_type_and_reshape),
+        Field(
+            description="A numpy array containing the input data.",
+        ),
+    ]
+    explanations: Annotated[
+        shap.Explanation,
+        Field(
+            description="A numpy array containing the SHAP explanations.",
+        ),
+    ]
+    explanation_method: Annotated[
+        str | None,
+        Field(
+            description="The method used to generate the explanation.",
+        ),
+    ] = None
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    def _validate_shapes(self):
+        """
+        Validates that the shape of the explanations matches the shape of the input data.
+
+        This method checks if the shape of `self.explanations` matches the shape of `self.data`.
+        If the shapes do not match, it checks if the explanation is multitarget - the last dimension of the explanation will not match the data shape.
+        In case the explanation is local - the explanation is one axis shorter, it unsquezees it - adds a new axis in the beginning of the explanation.
+        Otherwise, it raises a `ShapeMismatchError`
+        Raises:
+            ShapeMismatchError: If the shape of the explanations does not match the shape of the input data, the mismatch is not only in the last dimension and cannot unsqueeze the data, in case the explanation is local.
+        """
+        data_shape = self.data.shape
+        explanation_shape = self.explanations.shape
+        if explanation_shape == data_shape:
+            return  # no need to validate
+
+        if len(explanation_shape) == len(data_shape) + 1:
+            # the explanation has one more dimension than the data - the multitarget explanation, no local explanation
+            if explanation_shape[:-1] == data_shape:
+                logger.debug("Detected explanation for multiple targets based on the explanation shape validation.")
+                return
+            else:
+                raise ShapeMismatchError(
+                    f"Shape of the explanations does not match the shape of the input data. Expected {data_shape}, but got {self.explanations.shape}"
```

Comment:
If we want to be ideal we should remove fstring in errors: https://docs.astral.sh/ruff/rules/f-string-in-exception/, https://www.reddit.com/r/learnpython/comments/1eix0cy/why_avoid_using_fstrings_in_loggingerror_messages/?rdt=36416
```suggestion
                    "Shape of the explanations does not match the shape of the input data. Expected {data_shape}, but got {}".format(data_shape, self.explanations.shape)
```

Reply:
i would prefer to make it a separate pull request, since the fstring formatting is more common in our codebase - not only the shap module has this type of strings. 

But good remark though, thanks for keeping an eye on it